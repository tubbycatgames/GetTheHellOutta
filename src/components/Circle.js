import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components/native";

import Vector from "./Vector";

const getDiameter = ({ radius }) => radius * 2;
const getRadius = ({ radius }) => radius;

export const Circle = styled.View`
  background-color: ${props => props.color};
  border-color: white;
  border-radius: ${getDiameter};
  border-width: 1;
  height: ${getDiameter};
  left: ${props => props.position.x - getRadius(props)};
  position: absolute;
  top: ${props => props.position.y - getRadius(props)};
  width: ${getDiameter};
`;

Circle.propTypes = {
  color: PropTypes.string.isRequired,
  position: PropTypes.instanceOf(Vector).isRequired,
  radius: PropTypes.number.isRequired
};

const CircleWithRadius = (radius, displayName) => {
  const SizedCircle = props => <Circle radius={radius} {...props} />;
  SizedCircle.displayName = displayName;
  SizedCircle.propTypes = {
    color: PropTypes.string.isRequired,
    position: PropTypes.instanceOf(Vector).isRequired
  };
  return SizedCircle;
};

export const SmallCircle = CircleWithRadius(10, "SmallCircle");
export const MediumCircle = CircleWithRadius(20, "MediumCircle");
