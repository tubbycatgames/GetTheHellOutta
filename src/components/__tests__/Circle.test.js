import React from "react";
import renderer from "react-test-renderer";

import { Circle } from "../Circle";

describe("Circle", () => {
  it("should render a circle", () => {
    const circle = renderer.create(
      <Circle color="#543210" position={{ x: 5, y: 5 }} radius={5} />
    );
    expect(circle.toJSON()).toMatchSnapshot();
  });
});
