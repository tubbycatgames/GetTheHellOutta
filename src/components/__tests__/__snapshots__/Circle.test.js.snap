// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should render a circle 1`] = `
<View
  color="#543210"
  position={
    Vector {
      "x": 5,
      "y": 5,
    }
  }
  radius={5}
  style={
    Array [
      Object {
        "backgroundColor": "#543210",
        "borderColor": "white",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 10,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 10,
      },
    ]
  }
/>
`;
